File: src/runtime/julia/std/glob_native.jl
# Native implementation for pytra.std.glob
# Generated std/glob.jl delegates host bindings through this native seam.

module __GlobNative
    function glob(pattern)
        return String[f for f in readdir(dirname(pattern) == "" ? "." : dirname(pattern); join=true) if occursin(Regex(replace(replace(basename(pattern), "." => "\\."), "*" => ".*")), basename(f))]
    end
end

File: src/runtime/go/std/glob_native.go
// glob_native.go: @extern delegation for pytra.std.glob.
// Hand-written native implementation.
package main

import "path/filepath"

func glob(pattern string) []string {
	items, err := filepath.Glob(pattern)
	if err != nil {
		panic(err)
	}
	return items
}
func py_glob(pattern string) []string { return glob(pattern) }
